server/src/index.js

```diff
@@ -1,14 +1,19 @@
 const express = require('express');
 const { graphqlHTTP } = require('express-graphql');
-const typeDefs = require("./schema")
+const typeDefs = require("./schema/schema")
 const resolvers = require("./resolvers/resolvers")
+const schema = require("./schema/schema")
+const connectDB = require('./config/db')
+require('dotenv').config()
 
 const app = express();
 
+// Connection to the database
+connectDB();
+
 // Middleware to implement GraphQL
 app.use("/gql", graphqlHTTP({
-  schema: typeDefs,
-  rootValue: resolvers,
+  schema: schema,
   graphiql: true
 }))
 
```

Comment:
mongoose and mongo do not have immediate connections. 
In certain cases the initial connection might take longer, do not start the `.listen` on the server unless connected. 


provide `connectDB()` with a callback that waits for `mongoose.connection.on("connected",callback)` or add a `await mongoose.connect()` in your db file to actually let the connection finish and await that in this file as well.